python/lsst/db/db.py

```diff
@@ -54,547 +41,111 @@
-            if e.errCode() == DbException.TB_DOES_NOT_EXIST and not mustExist:
-                self._log.debug("dropTable failed, mustExist is False")
+class Db(object):
+    """@brief Thin wrapper around SQLAlchemy. Caches engine."""
+
+    def __init__(self, driverName="mysql", logFlag=False):
+        """
+        Initializes Db object.
+
+        @param driverName Name of the driver. Currently supported: "mysql"
+
+        @param logFlag    Flag turning on/off SQLAlchemy logging.
+        """
+        if driverName != "mysql":
+            raise DbException(DbException.UNSUPPORTED_DRIVER, driverName)
+        self._driverName = driverName
+        self._logFlag = logFlag
+
+    def getEngineFromFile(self, fileName):
+        """
+        Initializes and returns SQLAlchemy engine using values provided through the
+        file.
+
+        @param fileFile  File from which default client values are read. Only the
+                         values from section labeled as the driverName.
+
+        Raises DbException if fileName does not exist.
+        """
+        if fileName.startswith('~'):
+            fileName = os.path.expanduser(fileName)
+        if not os.path.isfile(fileName):
+            raise DbException(DbException.MISSING_OPT_FILE, fileName)
+        config = ConfigParser.ConfigParser()
+        config.readfp(open(fileName))
+        if self._driverName == "mysql":
+            args = self._processMySqlFile(fileName, config)
+        return self.getEngineFromArgs(**args)
+
+    def getEngineFromArgs(self, **kwArgs):
+        """
+        Initializes and returns SQLAlchemy engine using values provided through the
+        kwArgs. List of typically passed values:
+        - user or username
+        - passwd or password
+        - host
+        - port
+        - db or database
+        - query
+        To specify socket, use query={"unix_socket": "/the/socket.file"}
+        """
+        args = dict()
+        args["drivername"] = self._driverName
+        if "user" in kwArgs:
+            args["username"] = kwArgs["user"]
+        elif "username" in kwArgs:
+            args["username"] = kwArgs["username"]
+        if "passwd" in kwArgs:
+            args["password"] = kwArgs["passwd"]
+        elif "password" in kwArgs:
+            args["password"] = kwArgs["password"]
+        if "host" in kwArgs:
+            args["host"] = kwArgs["host"]
+        if "port" in kwArgs:
+            args["port"] = kwArgs["port"]
+        if "db" in kwArgs:
+            args["database"] = kwArgs["db"]
+        elif "database" in kwArgs:
+            args["database"] = kwArgs["database"]
+        if "query" in kwArgs:
+            args["query"] = kwArgs["query"]
+        url = URL(**args)
+        return sqlalchemy.create_engine(url, echo=self._logFlag)
+
+    def _addToQueryDict(self, theDict, theKey, theValue):
+        if "query" not in theDict:
+            theDict["query"] = dict()
+        theDict["query"][theKey] = theValue
```

Comment:
one-line replacement (it's idiomatic and I think it's more readable):

    theDict.setdefault('query', {})[theKey] = theValue